lib/main/atom/codefix/codeActionsProvider.ts

```diff
@@ -0,0 +1,51 @@
+import {ClientResolver} from "../../../client/clientResolver"
+import {compact, flatten, debounce} from "lodash"
+import {CompositeDisposable} from "atom"
+import {ErrorPusher} from "../../errorPusher"
+import {getEditorPosition, spanToRange, pointToLocation} from "../utils"
+import {GetTypescriptBuffer} from "../commands/registry"
+import {TypescriptServiceClient} from "../../../client/client"
+import {CodefixProvider} from "./codefixProvider"
+
+export interface Message {
+  filePath: string
+  range?: TextBuffer.IRange
+  // this interface is rater obviously incomplete
+}
+
+export interface CodeAction {
+  apply(): Promise<void>
+  getTitle(): Promise<string>
+  dispose(): void
+}
+
+export interface CodeActionProvider {
+  grammarScopes: string[]
+  priority: number
+  getCodeActions(
+    editor: AtomCore.IEditor,
+    range: TextBuffer.IRange,
+    diagnostics: Message[],
+  ): Promise<CodeAction[]>
+}
+
+export class CodeActionsProvider implements CodeActionProvider {
+  public grammarScopes = ["source.ts", "source.tsx"]
+  public priority = 0
+
+  constructor(private codefixProvider: CodefixProvider) {}
+
+  async getCodeActions(
+    textEditor: AtomCore.IEditor,
+    range: TextBuffer.IRange,
+    diagnostics: Message[],
+  ): Promise<CodeAction[]> {
+    return this.codefixProvider.runCodeFix(textEditor, range.start, fix => ({
+      getTitle: async () => fix.description,
```

Comment:
Neither of the two methods here need to be `async`.

Reply:
Actually they do, by the letter of the interface. I could use `Promise.resolve` of course, but I find this to be more succinct. Although you're right in that I should probably `await` `applyFix` below.

Reply:
Gotcha. Yeah, I also prefer using `async` over `Promise.resolve` even if it's a little less efficient.